coop_shift/report/report_wallchart_ftop.py

```diff
@@ -94,12 +94,12 @@ def _get_report_info(self, data):
             dates = [
                 datetime.strftime(h['date'], "\'%Y-%m-%d\'") for h in header]
 
-            sql = """SELECT begin_time, end_time
-                FROM shift_shift as ss, shift_type as st
-                WHERE date_without_time IN (%s) AND ss.shift_type_id = st.id
-                AND st.is_ftop = false
-                GROUP BY begin_time, end_time
-                ORDER BY begin_time""" % ",".join(dates)
+            sql = 'SELECT begin_time, end_time '\
+                'FROM shift_shift as ss, shift_type as st '\
+                'WHERE date_without_time IN (%s) AND ss.shift_type_id = st.id '\
+                'AND st.is_ftop = false '\
+                'GROUP BY begin_time, end_time '\
+                'ORDER BY begin_time' % ",".join(dates)
```

Comment:
You don't need \ in python to concatenate multiple lines.

You can do this:
```py
sql = 'SELECT begin_time, end_time '
         'FROM shift_shift as ss, shift_type as st '
         'WHERE date_without_time IN (%s) AND ss.shift_type_id = st.id '
         'AND st.is_ftop = false '
         'GROUP BY begin_time, end_time '
         'ORDER BY begin_time'
```
and it will work. But for readabily, when using long sqls.. this might be better:

```py
sql = """
    SELECT begin_time, end_time
    FROM shift_shift as ss, shift_type as st 
    WHERE date_without_time IN (%s) AND ss.shift_type_id = st.id
    AND st.is_ftop = false
    GROUP BY begin_time, end_time
    ORDER BY begin_time
""" % ','.join(dates)
```

But also here you have an SQL injection issue. Notice how you're inserting query using the str format function?
In this case it's probably safe because you know that dates will never be a malintencioned query. But it's a good practice to always go throught psycopg to send the parameters, like this:

```py
sql = """
    SELECT begin_time, end_time
    FROM shift_shift as ss, shift_type as st 
    WHERE date_without_time IN %s AND ss.shift_type_id = st.id
    AND st.is_ftop = false
    GROUP BY begin_time, end_time
    ORDER BY begin_time
"""

self.env.cr.execute(sql, (tuple(dates),))
```

There's more information about sql injection here: https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#id36